### PR TITLE
Fix deprecated warning from chrono

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

File: src/array/list/mutable.rs [hunk 1]
```diff
@@ -181,6 +181,7 @@ impl<O: Offset, M: MutableArray> MutableListArray<O, M> {
     /// - the new offsets are not in monotonic increasing order.
     /// - any new offset is not in bounds of the backing array.
     /// - the passed iterator has no upper bound.
+    #[allow(dead_code)]
```
Comment (Xuanwo) [inline src/array/list/mutable.rs:184]: I'm not sure do we need `extend_offsets` and `unsafe_extend_offsets` anymore.

Cc @jorgecarleitao for comments.

Comment (jorgecarleitao) [reply src/array/list/mutable.rs:184]: I believe that this is only used when json feature is active.

Comment (Xuanwo) [reply src/array/list/mutable.rs:184]: Since clippy didn't test with the `json` feature enabled, It's OK to add a `#[allow(dead_code)]` here?

**Comment by benesch** [comment]
Random thought: you may want to consider checking Cargo.lock into version control! Even though standard practice is to avoid checking in Cargo.lock for library crates, I've found it helpful in my library-only projects, as it ensures that new releases of dependencies can't break CI in my repository.

**Comment by Xuanwo** [comment]
> it ensures that new releases of dependencies can't break CI in my repository.

Interesting idea!

I will leave this for @jorgecarleitao to decide, although I prefer to break CI early. Checkout `Cargo.lock` in the repo will not change the fact that we need to resolve the deprecate warnings by hand, it only changes the time of finding them.

**Comment by jorgecarleitao** [comment]
I also think it is beneficial to have the CI issues flagged sooner than later.

**Comment by jorgecarleitao** [comment]
Thanks a lot for the PR, @Xuanwo !

**Pull request merged**